### ISSUE TITLE
Column renames not detected when adding other attributes to entity at the same time

### ISSUE THREAD

**Issue by sgronblo** [typeorm/typeorm#9222]
### Expected Behavior, Actual Behavior, Steps to Reproduce

I changed:

```
  @Column()
  description: string
```

to:

```
  @Column({ name: 'remark' })
  description: string
```

`typeorm schema:log` correctly identifies this as a column rename and generates the following migration:

```
ALTER TABLE "activity_group_erp_mapping" RENAME COLUMN "description" TO "remark";
```

However if I add a new attribute to the same entity:

```
  @Column({ nullable: true, type: 'int4' })
  siteId: number | null
```

Now `typeorm schema:log` shows the column changed as a destructive DROP and ADD:

```
ALTER TABLE "activity_group_erp_mapping" DROP COLUMN "description";
ALTER TABLE "activity_group_erp_mapping" ADD "siteId" integer;
ALTER TABLE "activity_group_erp_mapping" ADD "remark" character varying NOT NULL;
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |   macOS monterey 12.4 |
| Node.js version     | 16.13.0  | 
| Typescript version  | 4.7.4  | 
| TypeORM version     | 0.3.6  | 

### Relevant Database Driver(s)

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | no            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | yes            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✅ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by sgronblo** [comment]
After going back to my project. I noticed that even just trying to rename two attributes at the same time results in the same incorrect DROP & ADD combination.

**Comment by AlexMesser** [comment]
TL;DR: It is expected behavior.

From TypeORM side renaming looks like you just removed one column and added another. To differentiate add/remove from rename, TypeORM checks another column properties like type, isNullable, isUnique to make sure that this is the same column with only name changed. It works when you rename one column at time, but it's impossible to identify renamed columns when there are more than one columns changed at a time.
